### PR TITLE
Tenant claims test case

### PR DESCRIPTION
**Pull request by kavishkafernando**
## Purpose
Test case to add a new claim to tenant, check display order of claims and sign up new user.

### CODE REVIEW

File: modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/clients/APIStoreRestClient.java [hunk 1]
```diff
@@ -1271,6 +1271,27 @@ public HttpResponse signUp(String userName, String password, String firstName, S
             throw new APIManagerIntegrationTestException("Error in user sign up. Error: " + e.getMessage(), e);
         }
     }
+    /**
+     * API Store sign up
+     *
+     * @param userName - store user name
+     * @param password -store password
```
Comment (harsha89) [inline modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/clients/APIStoreRestClient.java]: slight issue with - need to have a space

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+
+package org.wso2.am.integration.tests.other;
+
+import com.google.gson.Gson;
+import org.apache.axis2.AxisFault;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.testng.annotations.*;
```
Comment (harsha89) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: remove wild card annotations

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+import org.wso2.carbon.identity.claim.metadata.mgt.stub.ClaimMetadataManagementServiceStub;
+import org.wso2.carbon.identity.claim.metadata.mgt.stub.dto.AttributeMappingDTO;
+import org.wso2.carbon.identity.claim.metadata.mgt.stub.dto.ClaimPropertyDTO;
+import org.wso2.carbon.identity.claim.metadata.mgt.stub.dto.LocalClaimDTO;
+import org.wso2.carbon.registry.resource.stub.ResourceAdminServiceExceptionException;
+import org.wso2.carbon.utils.FileManipulator;
+import org.wso2.carbon.utils.ServerConstants;
+
+import javax.xml.xpath.XPathExpressionException;
+import java.io.*;
```
Comment (harsha89) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: same here

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+import java.net.URL;
+import java.net.URLConnection;
+import java.rmi.RemoteException;
+import java.util.*;
```
Comment (harsha89) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: same here and follow the smae for other places

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+            JSONObject signUpJsonObject = null;
+            signUpJsonObject = new JSONObject(storeSignUpResponse.getData());
+            assertFalse(signUpJsonObject.getBoolean("error"), "Error in user sign up Response");
+            //login with new user
+            HttpResponse loginResponse = apiStore.login(STORE_USERNAME, STORE_PASSWORD);
+            JSONObject loginJsonObject = new JSONObject(loginResponse.getData());
+            assertFalse(loginJsonObject.getBoolean("error"), "Error in Login Request: User Name : " + STORE_USERNAME);
+
+        } catch (XPathExpressionException e) {
+            log.error("Error while getting store context");
```
Comment (harsha89) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: log with e

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+        } catch (JSONException e) {
+            log.error("Error while signing up");
```
Comment (harsha89) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: log with e

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+    /**
+     * This method replaces the signup.xml
+     * @throws APIManagerIntegrationTestException - exception throws when reading the file.
+     */
+    public void replaceSignUpXml() throws APIManagerIntegrationTestException {
+        initialize();
+        try {
+            resourceAdminServiceClient.updateTextContent(SIGNUP_XML_REG_CONFIG_LOCATION, newSignUPXML);
+        } catch (RemoteException e) {
+            log.error("Error while updating text context");
```
Comment (harsha89) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: log with e and follow same practice

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java [hunk 8]
```diff
@@ -0,0 +1,344 @@
...
+            }
+            return stringBuilder.toString();
+        } catch (IOException ioE) {
+            throw new APIManagerIntegrationTestException("IOException when reading the file from:" + fileLocation, ioE);
+        } finally {
+            if (bufferedReader != null) {
+                try {
+                    bufferedReader.close();
+                } catch (IOException e) {
+                    log.warn("Error when closing the buffer reade which used to reed the file:" + fileLocation +
```
Comment (harsha89) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: why this is a wan?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java [hunk 9]
```diff
@@ -0,0 +1,344 @@
...
+                    assertTrue(userFieldNames.get(4).contentEquals("Email"));
+                    assertTrue(userFieldNames.get(5).contentEquals("Telephone"));
+                    assertTrue(userFieldNames.get(6).contentEquals("Mobile"));
+                    assertTrue(userFieldNames.get(7).contentEquals("IM"));
+                    assertTrue(userFieldNames.get(8).contentEquals("URL"));
+                    assertTrue(userFieldNames.get(9).contentEquals("Department"));
+                    assertTrue(userFieldNames.get(10).contentEquals("Birth Date"));
+                    assertTrue(userFieldNames.get(11).contentEquals("Role"));
+                }
+                in.close();
```
Comment (harsha89) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: better to close in finally block

File: modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/clients/APIStoreRestClient.java (hunk 1 above)
Comment (kavishkafernando) [reply modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/clients/APIStoreRestClient.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java (hunk 2 above)
Comment (kavishkafernando) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java (hunk 3 above)
Comment (kavishkafernando) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java (hunk 4 above)
Comment (kavishkafernando) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java (hunk 5 above)
Comment (kavishkafernando) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java (hunk 6 above)
Comment (kavishkafernando) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java (hunk 7 above)
Comment (kavishkafernando) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java (hunk 9 above)
Comment (kavishkafernando) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: fixed

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java (hunk 8 above)
Comment (kavishkafernando) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/TenantClaimsTestCase.java]: changed it to error - fixed

**Pull request merged**